src/SiteMaster/Core/Auditor/QATestController.php

```diff
@@ -0,0 +1,64 @@
+<?php
+namespace SiteMaster\Core\Auditor;
+
+use SiteMaster\Core\Auditor\Site\Page;
+use SiteMaster\Core\Config;
+use SiteMaster\Core\Controller;
+use SiteMaster\Core\Registry\Registry;
+use SiteMaster\Core\Registry\Site;
+use SiteMaster\Core\User\Session;
+use SiteMaster\Core\ViewableInterface;
+use SiteMaster\Core\Auditor\Scan;
+
+class QATestController implements ViewableInterface
+{
+    public $url;
+    public $site;
+    public $scan;
+    public $page;
+    public $current_user;
+    
+    public function __construct($options = array())
+    {
+        if (!isset($_SERVER['HTTP_REFERER'])) {
+            throw new \Exception('An HTTP referer is missing! We don\'t know what you want to test.', 400);
+        }
+        
+        $this->url = $_SERVER['HTTP_REFERER'];
+
+        $registry = new Registry();
+        $this->site = $registry->getClosestSite($this->url);
+        
+        if ($this->site) {
+            //only find a scan if we found a site
+            $this->scan = $this->site->getLatestScan();
+        }
+        
+        if ($this->scan) {
+            //Only try to find a page scan if we found a site scan
+            $this->page = Page::getByScanIDAndURI($this->scan->id, $this->url);
+        }
+        
+        if ($this->page) {
+            //This page has been scanned, let's redirect to it.
+            Controller::redirect($this->page->getURL());
```

Comment:
If you just redirect an unsuspecting use to a page that is more tailored for someone that has more intimate knowledge of our tool, the user is probably going to be very confused. Maybe a FlashBagMessage, or some other kind of introduction to what you're taking then to.